ui/src/timeMachine/selectors/index.ts

```diff
@@ -249,3 +250,63 @@ export const getSaveableView = (state: AppState): QueryView & {id?: string} => {
 
   return saveableView
 }
+
+export const getStartTime = (timeRange: TimeRange): number => {
+  if (!timeRange) {
+    return Infinity
+  }
+  const {lower} = timeRange
+  switch (lower) {
+    case 'now() - 30d':
+      return moment()
+        .subtract(30, 'days')
+        .valueOf()
+    case 'now() - 7d':
+      return moment()
+        .subtract(7, 'days')
+        .valueOf()
+    case 'now() - 2d':
+      return moment()
+        .subtract(2, 'days')
+        .valueOf()
+    case 'now() - 24h':
+      return moment()
+        .subtract(24, 'hours')
+        .valueOf()
+    case 'now() - 12h':
+      return moment()
+        .subtract(12, 'hours')
+        .valueOf()
+    case 'now() - 6h':
+      return moment()
+        .subtract(6, 'hours')
+        .valueOf()
+    case 'now() - 1h':
+      return moment()
+        .subtract(1, 'hours')
+        .valueOf()
+    case 'now() - 15m':
+      return moment()
+        .subtract(15, 'minutes')
+        .valueOf()
+    case 'now() - 5m':
+      return moment()
+        .subtract(5, 'minutes')
+        .valueOf()
+    default:
+      return moment(lower).valueOf()
+  }
+}
+
+export const getEndTime = (timeRange: TimeRange): number => {
+  if (!timeRange) {
+    return null
+  }
+  const {upper} = timeRange
+  switch (upper) {
+    case null:
+      return null
+    default:
+      return moment(upper).valueOf()
+  }
+}
```

Comment:
you know what i'm gonna say about this right?

add some tests to `getStartTime` and `getEndTime`. should be very straightforward and will do a few things:

1. clarify your thinking on this by forcing you to think through the cases these functions handle
1. document the behavior of these functions
1. cover us against regressions in this behavior
1. reveal any small bugs that might have slipped through the cracks

i know you know all this, but it's valuable to reiterate it from time to time :)